content/en/docs/concepts/overview/kubernetes-api.md

```diff
@@ -22,21 +22,130 @@ external components communicate with one another.
 The Kubernetes API lets you query and manipulate the state of API objects in Kubernetes
 (for example: Pods, Namespaces, ConfigMaps, and Events).
 
-Most operations can be performed through the
-[kubectl](/docs/reference/kubectl/) command-line interface or other
-command-line tools, such as
-[kubeadm](/docs/reference/setup-tools/kubeadm/), which in turn use the
-API. However, you can also access the API directly using REST calls.
+Most operations can be performed through the [kubectl](/docs/reference/kubectl/)
+command-line interface or other command-line tools, such as
+[kubeadm](/docs/reference/setup-tools/kubeadm/), which in turn use the API.
+However, you can also access the API directly using REST calls. Kubernetes
+provides a set of [client
+libraries](/docs/reference/using-api/client-libraries/) for those looking to
+write applications using the Kubernetes API. 
+
+Kubernetes offers two "machine readable" APIs for listing all other Kubernetes
+APIs. Both are valuable tools for developers who are building client libraries,
+tooling, IDE features, documentation, and other tools interacting with
+Kubernetes.
+
+[Discovery](#discovery-api) is an API that provides information about available API names,
+resources, versions, and operations supported. This is a Kubernetes specific term. It is intended to be a small
+document providing a summary of available resources and does not detail specific
+schema for the resources. For reference around schemas for resources, please
+refer to the OpenAPI document.
+
+The [OpenAPI]{#openapi-specification} provides full [OpenAPI v2.0 and 3.0 schemas](https://www.openapis.org/) for all Kubernetes API
+endpoints. OpenAPI v3 is the preferred method for accessing OpenAPI as it
+provides a more comprehensive and accurate view of the API. It includes all
+available API paths, as well as all resources consumed and produced for every
+operation on every endpoint. This also includes any extensibility components
+that a cluster supports. This document is significantly larger than Discovery,
```

Comment:
The Kubernetes OpenAPI document is significantly (is it comprehensive or complete) larger than the Discovery API.
The Discovery API is a lightweight alternative that provides a subset of information in the Kubernetes OpenAPI document.